### PR TITLE
[Refactor] Refactor embedded GCS config setting

### PR DESCRIPTION
**Pull request by supasate**
- Refactor photo service to use common middleware with video service
- Refactor GCS setting into a config file

### CODE REVIEW

File: src/services/photo/index.js [hunk 1]
```diff
@@ -46,10 +41,12 @@ function uploadToGCS(reqFile) {
...
     stream.on('finish', () => {
-      const publicUrl = getPublicUrl(gcsname);
+      const publicUrl = getGCSPublicUrl(gcsFileName, gcsConfig);
 
-      reqFile.cloudStorageObject = gcsname; // eslint-disable-line no-param-reassign
-      reqFile.cloudStoragePublicUrl = publicUrl; // eslint-disable-line no-param-reassign
+      /* eslint-disable no-param-reassign */
+      reqFile.cloudStorageObject = gcsFileName;
+      reqFile.cloudStoragePublicUrl = publicUrl;
+      /* eslint-enable no-param-reassign */
```
Comment (parnurzeal) [inline src/services/photo/index.js:49]: Why is this comment below the actual code?

File: src/services/photo/index.js [hunk 2]
```diff
@@ -169,32 +165,43 @@ class PhotosService {
...
-  return uploadToGCSByUrl(url)
+function uploadSaveRespondByUrl(url, gcsConfig) {
+  return uploadToGCSByUrl(url, gcsConfig)
     .then((file) => savePhotoMetadata(file))
     .then((photoDoc) => respondWithPhotoMetadata(photoDoc))
     .catch((error) => Promise.reject(error));
 }
 
 class UploadPhotoFromUrlService {
+  setup(app) {
```
Comment (parnurzeal) [inline src/services/photo/index.js:185]: Sorry, I am not knowing class in Javascript that well.
Could you explain why we need setup(app)?

File: src/services/photo/index.js (hunk 1 above)
Comment (supasate) [inline src/services/photo/index.js:49]: It's a group comment (disable and then enable).

File: src/services/photo/index.js (hunk 2 above)
Comment (supasate) [inline src/services/photo/index.js:185]: Actually, `setup(app)` is belong to feathers. We need to call `this.app.get()` to get config in other functions. So, we need to initially assign `app` to `this` in the `setup` function. (http://docs.feathersjs.com/services/readme.html#the-setup-method)

File: src/services/photo/index.js (hunk 1 above)
Comment (parnurzeal) [inline src/services/photo/index.js:49]: oh... I see.

File: src/services/photo/index.js (hunk 2 above)
Comment (parnurzeal) [inline src/services/photo/index.js:185]: Cool! Thx for very good explanation.

**Comment by supasate** [comment]
Close #97

**Pull request merged**